app/src/main/java/com/example/drawingactivity/MainActivity.kt

```diff
@@ -1,11 +1,70 @@
 package com.example.drawingactivity
 
-import androidx.appcompat.app.AppCompatActivity
+import android.annotation.SuppressLint
 import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.widget.Button
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
+
+// What do I need to do?
+// create a Dialog on Click and BottomSlide
+// Dialog contains Gradient, Color Info Views, and Hue Selector
+// Be able to click on the views and they update
+// be able to close and reopen the view with it retaining last state
+// be able to change orientation without losing state
+
+// will grab values from my instantiated classes here to pass into the viewModel
 
 class MainActivity : AppCompatActivity() {
+
+    private val viewModel = CustomColorViewModel()
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        findViewById<Button>(R.id.create_dialog).setOnClickListener {
+            colorPickerDialog()
+        }
+        val wheel = findViewById<ColorWheel>(R.id.colorWheele)
+
+        wheel.changeColor = { viewModel.setPrimaryColor(wheel.index) }
+
+        viewModel.onUpdate = {
+            Log.d("Pepperoni Primary Color", "${it.primaryColor}")
+            wheel.wheelColor = it.primaryColor
+        }
+    }
+
+    @SuppressLint("ClickableViewAccessibility")
```

Comment:
Clean up everything relating to the homework and just make these changes from the current ticket

Reply:
👍